lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, 
+              :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return unless @services.include?('Vehicle Registration')
+    return unless vehicle.is_a?(Vehicle)
+
+    update_plate_type_and_fees(vehicle)
+    set_registration_date(vehicle)
+    @registered_vehicles << vehicle
+  end
+
+  def administer_written_test(registrant)
+    return false unless @services.include?('Written Test')
+    return false unless registrant.permit? && registrant.age >=16
```

Comment:
For these 3 methods, since we're checking for all these conditions to be true, I think it's easier to read if we just put them all into one if statement instead of breaking them up into multiple guard clauses. So this would look like `if @services.include?("Written Test") && registrant.permit? && registrant.age >= 16` then put the logic inside the if statement. 

But, great job checking for all these conditions!